src/routes/signup/+page.server.ts

```diff
@@ -4,32 +4,47 @@ import type { Actions } from './$types'
 import { z, ZodError } from 'zod'
 import { fail, redirect } from '@sveltejs/kit'
 import { PrismaClientKnownRequestError } from '@prisma/client/runtime/library'
+import { recaptchaResponse } from '$lib/utils/recaptcha'
 
 export const actions: Actions = {
   default: async ({ request, cookies, url }) => {
     const fields = Object.fromEntries(await request.formData())
+    let recaptchaResult
+
     try {
       const schema = z
         .object({
           name: z.string().min(1),
           email: z.string().email().toLowerCase(),
           password: z.string().min(6),
           confirmPassword: z.string().min(6),
+          'g-recaptcha-response': z.string(),
         })
         .refine((data) => data.password === data.confirmPassword, {
           message: "Passwords don't match",
           path: ['confirmPassword'],
         })
         .parse(fields)
 
-      const user = await createUser(schema.name, schema.email, schema.password)
+      recaptchaResult = await recaptchaResponse(schema['g-recaptcha-response'])
 
-      await createUserSessionAndCookie(user.id, cookies)
+      if (
+        !schema['g-recaptcha-response'] ||
+        (recaptchaResult.success && recaptchaResult.score > 0.5)
```

Comment:
i would move this two conditions into the function `recaptchaResponse` so you just return a boolean from there, that could be named `isRequestValid` or something instead of recaptchaResult.

Reply:
what about the score?

Reply:
Will do!